packages/neon/neon/lib/src/widgets/list_view.dart

```diff
@@ -1,66 +1,79 @@
     this.topScrollingChildren,
     super.key,
   });
 
-  final List<T>? items;
   final bool isLoading;
-  final dynamic error;
-  final Future Function() onRefresh;
-  final Widget Function(BuildContext, T data) builder;
+  final Object? error;
+  final RefreshCallback onRefresh;
   final String? scrollKey;
-  final bool withFloatingActionButton;
   final List<Widget>? topFixedChildren;
   final List<Widget>? topScrollingChildren;
+  final Widget sliver;
 
   @override
-  Widget build(final BuildContext context) => RefreshIndicator(
-        onRefresh: onRefresh,
-        child: Column(
-          children: [
-            ...?topFixedChildren,
-            NeonLinearProgressIndicator(
-              margin: const EdgeInsets.symmetric(
-                horizontal: 10,
-                vertical: 5,
+  Widget build(final BuildContext context) {
+    final refreshIndicatorKey = GlobalKey<RefreshIndicatorState>();
+    final hasFloatingActionButton = Scaffold.maybeOf(context)?.hasFloatingActionButton ?? false;
+
+    return RefreshIndicator.adaptive(
+      key: refreshIndicatorKey,
+      onRefresh: onRefresh,
+      child: Scrollbar(
+        child: CustomScrollView(
+          key: scrollKey != null ? PageStorageKey<String>(scrollKey!) : null,
+          primary: true,
+          slivers: [
+            if (topFixedChildren != null)
+              SliverList.builder(
+                itemCount: topFixedChildren!.length,
+                itemBuilder: (final context, final index) => topFixedChildren![index],
               ),
-              visible: isLoading,
-            ),
-            Expanded(
-              child: Scrollbar(
-                child: ListView(
-                  primary: true,
-                  key: scrollKey != null ? PageStorageKey<String>(scrollKey!) : null,
-                  padding: withFloatingActionButton ? const EdgeInsets.only(bottom: 88) : null,
-                  children: [
-                    ...?topScrollingChildren,
-                    NeonException(
-                      error,
-                      onRetry: onRefresh,
-                    ),
-                    if (items != null) ...[
-                      for (final item in items!) ...[
-                        builder(context, item),
-                      ],
-                    ],
-                  ],
+            if (error != null)
+              SliverToBoxAdapter(
+                child: NeonException(
+                  error,
+                  onRetry: () async => refreshIndicatorKey.currentState!.show(),
                 ),
               ),
+            if (topScrollingChildren != null)
+              SliverList.builder(
+                itemCount: topScrollingChildren!.length,
+                itemBuilder: (final context, final index) => topScrollingChildren![index],
+              ),
+            SliverPadding(
+              padding: hasFloatingActionButton ? const EdgeInsets.only(bottom: 88) : EdgeInsets.zero,
```

Comment:
I wonder if we can calculate that padding value somehow as I find it ugly to just use a constant (I know I wrote this code :sweat_smile:)

Reply:
I don't think there is a better way :(

But I think the value should be 80:
- 56 FAB height
- 16 FAB padding
- 8 ListView Padding